### ISSUE TITLE
Error 0x80070003 unsolved after trying all solutions

### ISSUE THREAD

**Issue by LaysonShen** [microsoft/WSL#11416]
### Windows Version

Microsoft Windows [版本 10.0.22631.3374]

### WSL Version

2.1.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.146.1-2

### Distro Version

Ubuntu

### Repro Steps

1. wsl -l -o
2. wsl --install Ubuntu

### Expected Behavior

Steps to set username and passwd

### Actual Behavior

Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80070003
Error: 0x80070003 ???????????

Press any key to continue...

### Diagnostic Logs

[WslLogs-2024-04-03_14-21-45.zip](https://github.com/microsoft/WSL/files/14847763/WslLogs-2024-04-03_14-21-45.zip)
attached is the log

**Comment by LaysonShen** [comment]
I have tried every means to solve it, but just like what he met mentioned in #11019 ,it didnt work!

**Comment by LaysonShen** [comment]
According to #11019 , he solved this issue by reinstalling windows. But how to solve this issue without reinstalling?

**Comment by OneBlue** [comment]
@LaysonShen: Looking at the logs it might be because the ubuntu app is in a weird state. Does repairing / reinstalling the app help ?

**Comment by LaysonShen** [comment]
Thank you for your reply. I have reinstalled Ubuntu for several times, but it still didnt work. Moreover, other linux subsystem apps couldnt work neither.

在 2024-04-05 07:21:10，Blue ***@***.***> 写道：

@LaysonShen: Looking at the logs it might be because the ubuntu app is in a weird state. Does repairing / reinstalling the app help ?

**Comment by OneBlue** [comment]
Thank you @LaysonShen.

Let's try something else. Can you capture a TDD trace of ubuntu.exe ? 
You can find instructions on how to do that in [section 12)](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#12-collect-wslservice-time-travel-debugging-traces)

**Comment by LaysonShen** [comment]
![图片](https://github.com/microsoft/WSL/assets/69374332/e475f246-691e-4843-940e-3795eead82ec)
Thanks @OneBlue ! I have followed the instructions. However, I have to admit my failure while recording wslservice.exe and the warning info has been shown above.
Then I turned to record ubuntu.exe and this time I successfully record everything helpful.
[ubuntu01.zip](https://github.com/microsoft/WSL/files/15123620/ubuntu01.zip)

**Comment by OneBlue** [comment]
Thank you @LaysonShen. Did you capture ubuntu.exe's state from its startup ? Looking at the TDD trace, the trace starts when ubuntu.exe is already exiting so unfortunately it doesn't contain the repro information.

**Comment by divisa-developer** [comment]
I also have ran into this problem, I opened my own issue but haven't received response #11553 if there is anything that could be provided to help solve this I would like to contribute since I can't reinstall OS at the moment or something like that

**Comment by LaysonShen** [comment]
I tried what Blue has told me to capture the log of the subsystem but in vain when filtering sessions. I have to give it up LOL

在 2024-05-06 22:34:04，divisa-developer ***@***.***> 写道：

I also have ran into this problem, I opened my own issue but haven't received response #11553 if there is anything that could be provided to help solve this I would like to contribute since I can't reinstall OS at the moment or something like that

**Comment by divisa-developer** [comment]
> I tried what Blue has told me to capture the log of the subsystem but in vain when filtering sessions. I have to give it up LOL 在 2024-05-06 22:34:04，divisa-developer ***@***.***> 写道： I also have ran into this problem, I opened my own issue but haven't received response #11553 if there is anything that could be provided to help solve this I would like to contribute since I can't reinstall OS at the moment or something like that — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***>

I'm sorry to hear that. If it's worth anything I managed to pin down my problem to a permission issue in the installation of drivers for the Hyper-V devices used by the Virtual Machine Platform, a company wide policy suddenly dissalowed installation of drivers and this messed the Hyper-V devices, after asking the sys admin to modify the policy I could boot up distributions again without issue. I noticed this beacuse in my device manager I found 3 Unkown Devices under the category others, when I tried to update their drivers it gave a message pointing to the permission issue.

I noticed your error message is a bit different than mine but it's worth a try.

**Comment by LaysonShen** [comment]
I ran into this error while using my own PC. Wish you good luck.

在 2024-05-06 23:13:37，divisa-developer ***@***.***> 写道：

I tried what Blue has told me to capture the log of the subsystem but in vain when filtering sessions. I have to give it up LOL 在 2024-05-06 22:34:04，divisa-developer @.> 写道： I also have ran into this problem, I opened my own issue but haven't received response #11553 if there is anything that could be provided to help solve this I would like to contribute since I can't reinstall OS at the moment or something like that — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: @.>
I'm sorry to hear that. If it's worth anything I managed to pin down my problem to a permission issue in the installation of drivers for the Hyper-V devices used by the Virtual Machine Platform, a company wide policy suddenly dissalowed installation of drivers and this messed the Hyper-V devices, after asking the sys admin to modify the policy I could boot up distributions again without issue. I noticed this beacuse in my device manager I found 3 Unkown Devices under the category others, when I tried to update their drivers it gave a message pointing to the permission issue.
I noticed your error message is a bit different than mine but it's worth a try.
